cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/guice/SourceControlModule.java

```diff
@@ -14,21 +14,33 @@
  * the License.
  */
 
-package io.cdap.cdap.sourcecontrol;
+package io.cdap.cdap.sourcecontrol.guice;
 
 import com.google.inject.AbstractModule;
 import com.google.inject.Scopes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
 import io.cdap.cdap.sourcecontrol.operationrunner.InMemorySourceControlOperationRunner;
+import io.cdap.cdap.sourcecontrol.operationrunner.RemoteSourceControlOperationRunner;
 import io.cdap.cdap.sourcecontrol.operationrunner.SourceControlOperationRunner;
 
 /**
  *  Guice module for source control management classes.
  */
 public class SourceControlModule extends AbstractModule {
 
+  private final CConfiguration cConf;
+
+  public SourceControlModule(CConfiguration cConf) {
+    this.cConf = cConf;
+  }
   @Override
   protected void configure() {
-    // TODO: CDAP-20322 add the remoteSourceControlOperationRunner
-    bind(SourceControlOperationRunner.class).to(InMemorySourceControlOperationRunner.class).in(Scopes.SINGLETON);
+    boolean workerPoolEnabled = cConf.getBoolean(Constants.TaskWorker.POOL_ENABLE);
+    if (workerPoolEnabled) {
+      bind(SourceControlOperationRunner.class).to(RemoteSourceControlOperationRunner.class).in(Scopes.SINGLETON);
+    } else {
+      bind(SourceControlOperationRunner.class).to(InMemorySourceControlOperationRunner.class).in(Scopes.SINGLETON);
+    }
```

Comment:
Just curious, why do we make these singletons?

Reply:
Actually I don't have a strong opinion if they should be singletons..

Reply:
removed singleton scope